andela/mmdp_site
# MMDP-260 Enable searching of events

**What does this PR do?**

Adds the ability to search events.

**Description of Task to be completed?**

As an administrator, I want to be able to search for a specific event on the 'Events' page by entering specific search criteria in a search box.

**How should this be manually tested?**

- Fetch and checkout to `ft-search-events-MMDP-260`.
- Run `yarn dev-server`.
- Open the frontend application and navigate to the events listing: `list-events`. You should see a list of all events or a prompt to add new events. If no events exist, add some with different titles.
- On the search input above the event, items type in a search string and click the search button. If your search matches the title of any of the events it will appear on a newly created listing that contains only the relevant results. If no event matches your search you will see no results.
- Erase all the search input and all events will appear.

**Any background context you want to provide?**

This solution accommodates/preserves pagination previously implemented on the page.

**What are the relevant Jira issues?**

[MMDP-260](https://viisaus.atlassian.net/browse/MMDP-260)

**Screenshots (if appropriate)**
#### Events listing
<img width="1086" alt="screenshot 2019-02-13 at 13 06 31" src="https://user-images.githubusercontent.com/8180548/52703974-9bf79180-2f90-11e9-9c81-eb17aa1848f9.png">

#### A search
<img width="1058" alt="screenshot 2019-02-13 at 13 07 40" src="https://user-images.githubusercontent.com/8180548/52703982-9f8b1880-2f90-11e9-898d-84cbb528d113.png">

#### Another search
<img width="1050" alt="screenshot 2019-02-13 at 13 08 16" src="https://user-images.githubusercontent.com/8180548/52703956-913cfc80-2f90-11e9-820b-8c947b14c6d3.png">

#### Yet another search
<img width="1055" alt="screenshot 2019-02-13 at 13 08 04" src="https://user-images.githubusercontent.com/8180548/52703970-97cb7400-2f90-11e9-8175-d969fda52e14.png">

**Questions:**
N/A
